synapse/storage/databases/main/event_push_actions.py

```diff
@@ -444,6 +445,20 @@ def _get_thread(thread_id: str) -> NotifCounts:
                 return main_counts
             return thread_counts.setdefault(thread_id, NotifCounts())
 
+        receipt_types_clause, receipts_args = make_in_list_sql_clause(
+            self.database_engine,
+            "receipt_type",
+            (ReceiptTypes.READ, ReceiptTypes.READ_PRIVATE),
+        )
+
+        # PostgreSQL and SQLite differ in comparing scalar numerics.
+        if isinstance(self.database_engine, PostgresEngine):
+            # GREATEST ignores NULLs.
+            receipt_stream_clause = "GREATEST(receipt_stream_ordering, ?)"
+        else:
+            # MAX returns NULL if any are NULL, so COALESCE to 0 first.
+            receipt_stream_clause = "MAX(COALESCE(receipt_stream_ordering, 0), ?)"
```

Comment:
Are these technically expressions? (clause, statement, predicate, expression 😵)

```suggestion
            receipt_stream_expression = "MAX(COALESCE(receipt_stream_ordering, 0), ?)"
```